flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveCatalogUtil.java

```diff
@@ -0,0 +1,337 @@
+	}
+
+	private static ColumnStatistics getColumnStatistics(
+			Map<String, CatalogColumnStatisticsDataBase> colStats,
+			StorageDescriptor sd,
+			ColumnStatisticsDesc desc) {
+		List<ColumnStatisticsObj> colStatsList = new ArrayList<>();
+
+		for (FieldSchema field : sd.getCols()) {
+			String hiveColName = field.getName();
+			String hiveColType = field.getType();
+
+			if (colStats.containsKey(hiveColName)) {
+				CatalogColumnStatisticsDataBase flinkColStat = colStats.get(field.getName());
+				ColumnStatisticsData statsData = getColumnStatisticsData(HiveCatalogUtil.convert(hiveColType), flinkColStat);
+				ColumnStatisticsObj columnStatisticsObj = new ColumnStatisticsObj(hiveColName, hiveColType, statsData);
+				colStatsList.add(columnStatisticsObj);
+			}
+		}
+
+		return new ColumnStatistics(desc, colStatsList);
+	}
+
+	/**
+	 * Create a map of Flink column stats from the given Hive column stats.
+	 */
+	public static Map<String, CatalogColumnStatisticsDataBase> createCatalogColumnStats(List<ColumnStatisticsObj> hiveColStats) {
+		Map<String, CatalogColumnStatisticsDataBase> colStats = new HashMap<>();
+
+		if (hiveColStats != null) {
+			for (ColumnStatisticsObj colStatsObj : hiveColStats) {
+				CatalogColumnStatisticsDataBase columnStats = createTableColumnStats(HiveCatalogUtil.convert(colStatsObj.getColType()), colStatsObj.getStatsData());
+
+				if (colStats != null) {
+					colStats.put(colStatsObj.getColName(), columnStats);
+				}
+			}
+		}
+
+		return colStats;
+	}
+
+	/**
+	 * Create Flink ColumnStats from Hive ColumnStatisticsData.
+	 * Note we currently assume that, in Flink, the max and min of ColumnStats will be same type as the Flink column type.
+	 * For example, for SHORT and Long columns, the max and min of their ColumnStats should be of type SHORT and LONG.
+	 */
+	private static CatalogColumnStatisticsDataBase createTableColumnStats(DataType colType, ColumnStatisticsData stats) {
+		if (stats.isSetBinaryStats()) {
+			BinaryColumnStatsData binaryStats = stats.getBinaryStats();
+			return new CatalogColumnStatisticsDataBinary(
+					binaryStats.getMaxColLen(),
+					binaryStats.getAvgColLen(),
+					binaryStats.getNumNulls());
+		} else if (stats.isSetBooleanStats()) {
+			BooleanColumnStatsData booleanStats = stats.getBooleanStats();
+			return new CatalogColumnStatisticsDataBoolean(
+					booleanStats.getNumFalses(),
+					booleanStats.getNumFalses(),
+					booleanStats.getNumNulls());
+		} else if (stats.isSetDateStats()) {
+			DateColumnStatsData dateStats = stats.getDateStats();
+			return new CatalogColumnStatisticsDataDate(
+					new org.apache.flink.table.catalog.stats.Date(dateStats.getLowValue().getDaysSinceEpoch()),
+					new org.apache.flink.table.catalog.stats.Date(dateStats.getHighValue().getDaysSinceEpoch()),
+					dateStats.getNumDVs(),
+					dateStats.getNumNulls());
+		} else if (stats.isSetDoubleStats()) {
+			if (colType.getLogicalType() instanceof FloatType ||
```

Comment:
I'm wondering why we need to check the column data type. As long as the stats object is double, we can convert it to Flink double regardless the column type. What is the potential issue?